### PR TITLE
Replace some RHCOS / RHEL hardcoded names

### PR DESCRIPTION
**Pull request by vrutkovs**
This ensures FCOS/Fedora is mentioned in OKD docs.

Ref: https://github.com/openshift/openshift-docs/issues/21293

### CODE REVIEW

**Comment by vrutkovs** [comment]
@mburke5678 PTAL

File: modules/installation-requirements-user-infra.adoc [hunk 1]
```diff
@@ -48,7 +48,7 @@ endif::ibm-z[]
 The bootstrap, control plane, and compute machines must use the {op-system-first} as the
 operating system.
 
-Note that {op-system} is based on Red Hat Enterprise Linux 8 and inherits all of its hardware certifications and requirements.
+Note that {op-system} is based on {op-system-base-full} and inherits all of its hardware certifications and requirements.
```
Comment (mburke5678) [inline modules/installation-requirements-user-infra.adoc:51]: @vrutkovs 
s/{op-system-base-full}/{op-system-base-full} 8

I think we need the RHEL version here.

Comment (vrutkovs) [reply modules/installation-requirements-user-infra.adoc:51]: Good catch. Added `8` there and hid it in OKD docs - FCOS is a rolling release and may switch to versions arbitrarily

Comment (mburke5678) [reply modules/installation-requirements-user-infra.adoc:51]: @vrutkovs I merged this PR but I'm not seeing the `8` in master. 
https://github.com/openshift/openshift-docs/blob/master/modules/installation-requirements-user-infra.adoc

Comment (vrutkovs) [reply modules/installation-requirements-user-infra.adoc:51]: Added it in https://github.com/openshift/openshift-docs/pull/21640

**Pull request merged**